### ISSUE TITLE
A new instance of the same WLED accessory is not recognized as the previous one

### ISSUE THREAD

**Issue by osxster** [goodshort/homebridge-wled-preset#10]
I am intermittently seeing an issue with WLED-preset if one of my wled controllers is crashed or off the network.  Occasionally WLED seems to crash and is unreachable.  If that happens, it reorders my WLED controllers in HomeKit.  I setup automations in HomeKit to turn the lights on at sunset.  When this problem happens HomeKit sees another WLED controller as the missing one and issues automation to that controller instead.  Rebooting the down WLED controller does not fix it.  The only fix is to restart homebridge and the WLED controller that was down gets discovered as a whole new controller which needs to be setup again in HomeKit.  I have five digiquads and they were all configured in WLED-preset.  The controllers have static ips so it is not an IP address problem.

**Comment by agoodshort** [comment]
Oh wow, just reproduced the issue, was not aware that Homebridge was literally crashing if a WLED accessory was going down. But good thing it restarts instantly, it does not crash the whole thing and have you restart manually. Log below:
```
[02/01/2022, 19:29:22] [WledPreset] Office Led Strip: Turning off
[02/01/2022, 19:29:56] Error: connect EHOSTUNREACH 192.168.1.20:80
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1161:16)
[02/01/2022, 19:29:56] Got SIGTERM, shutting down Homebridge...
[02/01/2022, 19:29:59] Error: connect EHOSTUNREACH 192.168.1.20:80
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1161:16)
[02/01/2022, 19:30:01] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[02/01/2022, 19:30:06] [HB Supervisor] Restarting Homebridge...
[02/01/2022, 19:30:06] [HB Supervisor] Started Homebridge v1.3.8 with PID: 13923
[02/01/2022, 19:30:09] Loaded config.json with 1 accessories and 5 platforms.
```

So there are 2 issues here:
1. Plugin crashes when WLED accessory disconnect
2. A new instance of the same WLED accessory is not recognize as the previous one

> When this problem happens HomeKit sees another WLED controller as the missing one and issues automation to that controller instead.

This is such a weird behaviour... I don't really understand how this happen.

My plans for this issue for now:
For the sack of simplicity I used IP addresses as unique identifier for the accessory instances. I will try to implement the MAC address instead to get a much more unique ID. I am not sure if this will resolve the issue of the automation being transfered to the next instance, but so far this is what I think could resolve this (and this was something I really to implement).

Thanks a lot @osxster for your continious bug/enhancement report. It is highly appreciated, I appologies if I am not working too often on this plugin.

**Comment by osxster** [comment]
Absolutely, I'm glad to help!  These WLED controllers seem to go down more often than I like, not sure if it is a WLED issue (well it really is), or if it could be being complicated by the plugin.  I have not looked at the code at all, but not sure if there could be any polling issues by the plugin that WLED doesn't like and it causes the IP stack to crash.  But when they go down, they stay running as my playlist still plays, but the controller is unreachable by IP.  While playlists are not supported by your plugin, because they stay running, I'm still using your plugin to automate turning on or off my controllers for the Christmas lights.

I do want to point out that if one of the WLED controllers goes down, the plugin is identifying another controller to take it's place.  So I have one WLED controller that I only use for testing light strips.  I've added it to the plugin, but that's it.  There are no automations on it configured in HomeKit.  It's just configured and assigned to a room in my HomeKit implementation.  Nothing else.  If one of my 'prod' WLED controllers goes down, any HomeKit automations begin occurring on that WLED controller that I was only using for testing of lights.  

So I think there is a 3rd issue in that if one WLED controller goes down, the plugin shouldn't elect another WLED controller to take it's place.

**Comment by agoodshort** [comment]
Quick update, I have done some good work these last few days and have some updates.
1. I will have a fix regarding homebridge crashing when a device goes down
2. I started to implement a mDNS/zeroconf approach where devices are automatically discovered (no need anymore to type them in the config file, you can still do so if you want to)

I hope these 2 new feature will fix the issues you are facing here. I will keep you posted here and I push the update to npm. I need a few more days to implement these properly.

**Comment by agoodshort** [comment]
Renaming the issue for clarity and to only point out one issue here as issue #14 just got created regarding the shutdown
